2024-06-10/server/main.py

```diff
@@ -0,0 +1,161 @@
+"""
+Coding challenge -- playing with APIs.
+I built this half, and Spencer wrote functions to communicate
+to the various endpoints on this API.
+- SAM 2024-06-10
+"""
+
+import json
+import os
+import inspect
+import uuid
+
+from fastapi import APIRouter, FastAPI
+
+app = FastAPI()
+router = APIRouter(prefix="/api")
+
+USERS = "users.txt"
+
+
+def read_users() -> dict:
+    with open(USERS, "r") as file_handle:
```

Comment:
What if file doesn't exist?

Reply:
Hey, good eye. Didn't catch this during testing because we hoisted the existence checking logic up into another function. Got this fixed in 7681f8ab1f761e1fa25e85505a7d69288e12b973.